src/components/NotFound.astro

```diff
@@ -130,9 +138,10 @@ const dictionary = Object.fromEntries(
 			try {
 				if (!this.dataset.dictionary) return;
 				const dictionary = JSON.parse(this.dataset.dictionary);
+        const languages = Object.keys(dictionary);
 
 				// Get current language from the URL path.
-				const langCodeMatch = location.pathname.match(/\/([a-z]{2}-?[a-z]{0,2})\//);
+				const langCodeMatch = location.pathname.match(`\\/(${languages.join('|')})\\/`);
```

Comment:
I don’t think this works? The argument to `match()` here is just treated as a string and tested for basic equality. And when constructing a regular expression from a string, you don’t need to escape the `/` either:

```suggestion
				const langCodeMatch = location.pathname.match(new RegExp('/(' + languages.join('|') + ')/'));
```

Reply:
The string gets converted into a regex by [`.match`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match#parameters). Ahh, yep. I think the `\\` was left over from when I was working on the inline regex. Made a commit to remove